Packs/rasterize/Integrations/rasterize/rasterize.py

```diff
@@ -365,39 +372,148 @@ def start_chrome_headless(chrome_port, chrome_binary=CHROME_EXE, user_options=""
+                demisto.log(f"chrome_options in chromes_options and instance_id not in instances_id (case 2)")
+            elif chrome_options not in chromes_options and instance_id not in instances_id:
+                demisto.log(f"chrome_options not in chromes_options and instance_id not in instances_id (case 3)")
+
+        return generate_new_chrome_instance(instance_id, chrome_options)
+
+    elif chrome_options not in instance_id_to_chromes_options[instance_id] and instance_id in instances_id:
+        """
+        When:
+            case 4: instance updated the chrome configuration to different configuration
+        Then:
+            should kill the old browser and create new one
+        """
+
+        demisto.log(f"case 4, chrome_options not in chromes_options and instance_id in instances_id")  # TODO: Remove log
+        demisto.debug(f"Found instance_id but with different {chrome_options=}")
+        chrome_port = instance_id_to_port.get(instance_id)
+        terminate_chrome(chrome_port)
+        delete_row_with_old_chrome_configurations_from_info_file(info, instance_id, chrome_port)
+        return generate_new_chrome_instance(instance_id, chrome_options)
+
+    elif chrome_options in instance_id_to_chromes_options[instance_id] and instance_id in instances_id:
+        """
+        When:
+            case 5: The instance exists in the file and the chrome configuration match to it
+        Then:
+            use the existing browser
+        """
+
+        demisto.log(f"case 5 - instance_id linked to chrome_options - using the existing chrome")  # TODO: Remove log
+        demisto.debug(f"Found {instance_id=} with matching {chrome_options=}")
+        chrome_port = chrome_options_to_port.get(chrome_options)
+        browser = is_chrome_running_locally(chrome_port)
+        return browser, chrome_port
 
+    else:
+        message = f"ERROR: Cannot connect to Chrome with parameters: {instance_id=}, {chrome_options=}"
+        demisto.info(message)
+        demisto.error(message)
+
+
+def get_chrome_instances_info_dictionaries(info):
+    instance_id_to_chromes_options = defaultdict(list)
+    chrome_options_to_port = {}
+    instance_id_to_port = {}
+
+    if info:
+        info_without_new_lines = info.strip()
+        lines = info_without_new_lines.splitlines()
+        for line in lines:
+            column = line.strip().split('\t')
+            port, instance_id, chrome_options = column
+            instance_id_to_chromes_options[instance_id].append(chrome_options)
+            chrome_options_to_port[chrome_options] = port
+            instance_id_to_port[instance_id] = port
+
+    return instance_id_to_chromes_options, chrome_options_to_port, instance_id_to_port
+
+
+def generate_new_chrome_instance(instance_id, chrome_options):
+    chrome_port = get_chrome_port()
+    browser, chrome_port = start_chrome_headless(str(chrome_port), instance_id, chrome_options)
+    return browser, chrome_port
+
+
+def get_chrome_port():
```

Comment:
Maybe rename to `generate_chrome_port`? After all - You're generating a new port, not getting an existing port.
```suggestion
def _chrome_port():
```